lib/musiclibrarycontroller.rb

```diff
@@ -48,14 +35,14 @@ def list_genres
   end
 
   def play_song
-    size = @control.files.length
+    size = Song.all.length
     Message.what_song(size)
-    index = gets.chomp.to_i
+    selection = gets.chomp.to_i - 1
 
-    if index < 1 || index > size
+    if [0..size].include?(selection) && selection
```

Comment:
**Code functionality**
How does this work? 

From what I can see, if the selection exists within the expected range and the selection value exists, an invalid song message is displayed.

Is this accurate?